utils/cache.py

```
@@ -15,21 +15,34 @@
-    It uses Prometheus-style counters and histograms to record submission attempts, cache interactions, and document processing metrics.
+    It uses Prometheus-style counters and histograms to record submission attempts, cache interactions metrics.
 
     Attributes:
         NAMESPACE (str): The base namespace for all metrics in the web application.
     """
 
     NAMESPACE = "dozens"
-
+    _instance = None
+
+    def __new__(cls):
```

Ruff (0.11.9)

Missing return type annotation for special method `__new__`

(ANN204)